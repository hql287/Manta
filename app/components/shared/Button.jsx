// Libraries
import React from 'react';
import PropTypes from 'prop-types';

// Styles
import styled from 'styled-components';

const ButtonStyle = styled.button`
  display: inline-flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  border: none;
  border-radius: 4px;
  padding: 4px 15px;
  font-size: 12px;
  text-decoration: none;
  background: #ffffff;
  border: 1px solid #e0e1e1;
  text-transform: uppercase;
  letter-spacing: 1px;
  // Block Level Button
  ${props => props.block && `width: 100%;`}
  // Color
  ${props => props.primary &&  `
    background: #469fe5;
    color: white;
  `}
  ${props => props.success && `
    background: #6bbb69;
    color: white;
  `}
  ${props => props.danger && `
    background: #EC476E;
    color: white;
  `}
  // Active state
  ${props => props.active && `
    background: #F2F3F4;
    color: #4F555C;
  `}
  // Hover
  &:hover {
    cursor: pointer;
    text-decoration: none;
    // color: white;
  }
`;

const ButtonLinkStyle = styled.button`
  display: flex;
  align-items: center;
  justify-content: center;
  background: none;
  border: none;
  padding: 0;
  margin: 0;
  ${props => props.primary && `color: #469fe5;`} ${props =>
  props.success && `color: #6bbb69;`} ${props =>
  props.danger && `color: #EC476E;`} &:hover {
    cursor: pointer;
  }
`;

const ButtonsGroupStyle = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: flex-start;
  > button {
    margin: 0!important;
    border-radius: 0;
    &:first-child { border-radius: 4px 0 0 4px; }
    &:last-child { border-radius: 0 4px 4px 0; }
    &:not(:first-child) { border-left: 0; }
  }
`;

function Button(props) {
  return props.link ? (
    <ButtonLinkStyle {...props}>{props.children}</ButtonLinkStyle>
  ) : (
    <ButtonStyle {...props}>{props.children}</ButtonStyle>
  );
}

Button.propTypes = {
  link: PropTypes.bool,
  danger: PropTypes.bool,
  onClick: PropTypes.func,
  primary: PropTypes.bool,
  success: PropTypes.bool,
};

Button.defaultProps = {
  link: false,
  primary: false,
  success: false,
  danger: false,
};

export const ButtonsGroup = props => (
  <ButtonsGroupStyle>{props.children}</ButtonsGroupStyle>
);

export default Button;
