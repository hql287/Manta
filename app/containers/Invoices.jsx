// Libs
import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { compose } from 'recompose';
import { connect } from 'react-redux';
const openDialog = require('../renderers/dialog.js');
const ipc = require('electron').ipcRenderer;
import { translate } from 'react-i18next';

// Actions
import * as Actions from '../actions/invoices';

// Selectors
import { getInvoices } from '../reducers/InvoicesReducer';
import { getDateFormat } from '../reducers/SettingsReducer';

// Components
import Invoice from '../components/invoices/Invoice';
import Message from '../components/shared/Message';
import _withFadeInAnimation from '../components/shared/hoc/_withFadeInAnimation';
import {
  PageWrapper,
  PageHeader,
  PageHeaderTitle,
  PageContent,
} from '../components/shared/Layout';

class Invoices extends PureComponent {
  constructor(props) {
    super(props);
    this.editInvoice = this.editInvoice.bind(this);
    this.deleteInvoice = this.deleteInvoice.bind(this);
    this.setInvoiceStatus = this.setInvoiceStatus.bind(this);
  }

  // Load Invoices & add event listeners
  componentDidMount() {
    // Add Event Listener
    ipc.on('confirmed-delete-invoice', (event, index, invoiceId) => {
      if (index === 0) {
        this.confirmedDeleteInvoice(invoiceId);
      }
    });
  }

  // Remove all IPC listeners when unmounted
  componentWillUnmount() {
    ipc.removeAllListeners('confirmed-delete-invoice');
  }

  // Open Confirm Dialog
  deleteInvoice(invoiceId) {
    const { t } = this.props;
    openDialog(
      {
        type: 'warning',
        title: t('dialog:deleteInvoice:title'),
        message: t('dialog:deleteInvoice:message'),
        buttons: [
          t('common:yes'),
          t('common:noThanks')
        ],
      },
      'confirmed-delete-invoice',
      invoiceId
    );
  }

  // Confirm Delete an invoice
  confirmedDeleteInvoice(invoiceId) {
    const { dispatch } = this.props;
    dispatch(Actions.deleteInvoice(invoiceId));
  }

  // set the invoice status
  setInvoiceStatus(invoiceId, status) {
    const { dispatch } = this.props;
    dispatch(Actions.setInvoiceStatus(invoiceId, status));
  }

  editInvoice(invoice) {
    const { dispatch } = this.props;
    dispatch(Actions.editInvoice(invoice));
  }

  // Render
  render() {
    const { dateFormat, invoices, t } = this.props;
    const invoicesComponent = invoices.map((invoice, index) => (
      <Invoice
        key={invoice._id}
        dateFormat={dateFormat}
        deleteInvoice={this.deleteInvoice}
        editInvoice={this.editInvoice}
        setInvoiceStatus={this.setInvoiceStatus}
        index={index}
        invoice={invoice}
        t={t}
      />
    ));
    return (
      <PageWrapper>
        <PageHeader>
          <PageHeaderTitle>{t('invoices:header:name')}</PageHeaderTitle>
        </PageHeader>
        <PageContent bare>
          {invoices.length === 0 ? (
            <Message info text={t('messages:noInvoice')} />
          ) : (
            <div className="row">{invoicesComponent}</div>
          )}
        </PageContent>
      </PageWrapper>
    );
  }
}

// PropTypes Validation
Invoices.propTypes = {
  dispatch: PropTypes.func.isRequired,
  invoices: PropTypes.arrayOf(PropTypes.object).isRequired,
  t: PropTypes.func.isRequired,
};

// Map state to props & Export
const mapStateToProps = state => ({
  invoices: getInvoices(state),
  dateFormat: getDateFormat(state),
});

export default compose(
  connect(mapStateToProps),
  translate(['common', 'form', 'invoices']),
  _withFadeInAnimation
)(Invoices);
