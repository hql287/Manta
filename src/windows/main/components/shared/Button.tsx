// Libraries
import React, { FC, ReactNode } from 'react'
import styled from 'styled-components'

interface ButtonProps {
  link?: boolean
  danger?: boolean
  onClick?: () => void
  primary?: boolean
  success?: boolean
  children: ReactNode
  block?: boolean
  active?: boolean
}

const ButtonStyle = styled.button<ButtonProps>`
  display: inline-flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  border: none;
  border-radius: 4px;
  padding: 4px 15px;
  font-size: 12px;
  text-decoration: none;
  background: #ffffff;
  border: 1px solid #e0e1e1;
  text-transform: uppercase;
  letter-spacing: 1px;
  ${(props) => props.block && `width: 100%;`}
  ${(props) =>
    props.primary &&
    `
    background: #469fe5;
    color: white;
  `}
  ${(props) =>
    props.success &&
    `
    background: #6bbb69;
    color: white;
  `}
  ${(props) =>
    props.danger &&
    `
    background: #EC476E;
    color: white;
  `}
  ${(props) =>
    props.active &&
    `
    background: #F2F3F4;
    color: #4F555C;
  `}
  &:hover {
    cursor: pointer;
    text-decoration: none;
  }
`

const ButtonLinkStyle = styled.button<ButtonProps>`
  display: flex;
  align-items: center;
  justify-content: center;
  background: none;
  border: none;
  padding: 0;
  margin: 0;
  ${(props) => props.primary && `color: #469fe5;`}
  ${(props) => props.success && `color: #6bbb69;`}
  ${(props) => props.danger && `color: #EC476E;`}
  &:hover {
    cursor: pointer;
  }
`

const ButtonsGroupStyle = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: flex-start;
  > button {
    margin: 0 !important;
    border-radius: 0;
    &:first-child {
      border-radius: 4px 0 0 4px;
    }
    &:last-child {
      border-radius: 0 4px 4px 0;
    }
    &:not(:first-child) {
      border-left: 0;
    }
  }
`

export const Button: FC<ButtonProps> = (props) => {
  return props.link ? (
    <ButtonLinkStyle {...props}>{props.children}</ButtonLinkStyle>
  ) : (
    <ButtonStyle {...props}>{props.children}</ButtonStyle>
  )
}

export const ButtonsGroup: FC<{ children: ReactNode }> = ({ children }) => (
  <ButtonsGroupStyle>{children}</ButtonsGroupStyle>
)
