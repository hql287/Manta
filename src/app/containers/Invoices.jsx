// Libs
import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { compose } from 'recompose';
import { connect } from 'react-redux';
const openDialog = require('../renderers/dialog.js');
const ipc = require('electron').ipcRenderer;
import { translate } from 'react-i18next';

// Actions
import * as Actions from '../actions/invoices';

// Selectors
import { getInvoices } from '../reducers/InvoicesReducer';
import { getDateFormat } from '../reducers/SettingsReducer';

// Components
import Invoice from '../components/invoices/Invoice';
import Message from '../components/shared/Message';
import Button, { ButtonsGroup } from '../components/shared/Button';
import _withFadeInAnimation from '../components/shared/hoc/_withFadeInAnimation';
import {
  PageWrapper,
  PageHeader,
  PageHeaderTitle,
  PageHeaderActions,
  PageContent,
} from '../components/shared/Layout';

export class Invoices extends PureComponent {
  constructor(props) {
    super(props);
    this.state = { filter: null };
    this.editInvoice = this.editInvoice.bind(this);
    this.deleteInvoice = this.deleteInvoice.bind(this);
    this.duplicateInvoice = this.duplicateInvoice.bind(this);
    this.setInvoiceStatus = this.setInvoiceStatus.bind(this);
    this.setFilter = this.setFilter.bind(this);
  }

  // Load Invoices & add event listeners
  componentDidMount() {
    // Add Event Listener
    ipc.on('confirmed-delete-invoice', (event, index, invoiceId) => {
      if (index === 0) {
        this.confirmedDeleteInvoice(invoiceId);
      }
    });
  }

  // Remove all IPC listeners when unmounted
  componentWillUnmount() {
    ipc.removeAllListeners('confirmed-delete-invoice');
  }

  // Open Confirm Dialog
  deleteInvoice(invoiceId) {
    const { t } = this.props;
    openDialog(
      {
        type: 'warning',
        title: t('dialog:deleteInvoice:title'),
        message: t('dialog:deleteInvoice:message'),
        buttons: [
          t('common:yes'),
          t('common:noThanks')
        ],
      },
      'confirmed-delete-invoice',
      invoiceId
    );
  }

  // Confirm Delete an invoice
  confirmedDeleteInvoice(invoiceId) {
    const { dispatch } = this.props;
    dispatch(Actions.deleteInvoice(invoiceId));
  }

  // set the invoice status
  setInvoiceStatus(invoiceId, status) {
    const { dispatch } = this.props;
    dispatch(Actions.setInvoiceStatus(invoiceId, status));
  }

  editInvoice(invoice) {
    const { dispatch } = this.props;
    dispatch(Actions.editInvoice(invoice));
  }

  duplicateInvoice(invoice) {
    const { dispatch } = this.props;
    dispatch(Actions.duplicateInvoice(invoice));
  }

  setFilter(event) {
    const currentFilter = this.state.filter;
    const newFilter = event.target.dataset.filter;
    this.setState({ filter: currentFilter === newFilter ? null : newFilter });
  }

  // Render
  render() {
    const { dateFormat, invoices, t } = this.props;
    const { filter } = this.state;
    const filteredInvoices = filter ? invoices.filter(invoice => invoice.status === filter) : invoices
    const invoicesComponent = filteredInvoices.map((invoice, index) => (
      <Invoice
        key={invoice._id}
        dateFormat={dateFormat}
        deleteInvoice={this.deleteInvoice}
        duplicateInvoice={this.duplicateInvoice}
        editInvoice={this.editInvoice}
        setInvoiceStatus={this.setInvoiceStatus}
        index={index}
        invoice={invoice}
        t={t}
      />
    ));
    // Filter Buttons
    const statuses = ['paid', 'pending', 'refunded', 'cancelled'];
    const filterButtons = statuses.map(status => (
      <Button
        key={`${status}-button`}
        active={filter === status}
        data-filter={status}
        onClick={this.setFilter}
      >
        { t(`invoices:status:${status}`) }
      </Button>
    ));

    return (
      <PageWrapper>
        <PageHeader>
          <PageHeaderTitle>{t('invoices:header:name')}</PageHeaderTitle>
          <PageHeaderActions>
            <i className="ion-funnel" />
            <ButtonsGroup>{ filterButtons }</ButtonsGroup>
          </PageHeaderActions>
        </PageHeader>
        <PageContent bare>
          {invoices.length === 0 ? (
            <Message info text={t('messages:noInvoice')} />
          ) : (
            <div className="row">
              {invoicesComponent}
            </div>
          )}
        </PageContent>
      </PageWrapper>
    );
  }
}

// PropTypes Validation
Invoices.propTypes = {
  dispatch: PropTypes.func.isRequired,
  invoices: PropTypes.arrayOf(PropTypes.object).isRequired,
  t: PropTypes.func.isRequired,
};

// Map state to props & Export
const mapStateToProps = state => ({
  invoices: getInvoices(state),
  dateFormat: getDateFormat(state),
});

export default compose(
  connect(mapStateToProps),
  translate(),
  _withFadeInAnimation
)(Invoices);
