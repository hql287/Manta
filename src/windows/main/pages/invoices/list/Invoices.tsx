// Libraries
import React, { useEffect, useState } from 'react'
import { create } from 'zustand'

const { ipcRenderer: ipc, createDialogWindow } = window.electronAPI

// Components
import Invoice from '@uiMainComponents/invoices/Invoice'
import Message from '@uiSharedComponents/Message'
import { Button, ButtonsGroup } from '@uiSharedComponents/Button'
import {
  PageWrapper,
  PageHeader,
  PageHeaderTitle,
  PageHeaderActions,
  PageContent,
} from '@uiSharedComponents/Layout'
import _withFadeInAnimation from '@uiSharedComponents/hoc/_withFadeInAnimation'

// Zustand Store
interface Invoice {
  _id: string
  status: string
}

interface InvoiceState {
  invoices: Invoice[]
  deleteInvoice: (invoiceId: string) => void
  setInvoiceStatus: (invoiceId: string, status: string) => void
  editInvoice: (invoice: Invoice) => void
  duplicateInvoice: (invoice: Invoice) => void
}

const useInvoiceStore = create<InvoiceState>((set) => ({
  invoices: [],
  deleteInvoice: (invoiceId) => {
    set((state) => ({
      invoices: state.invoices.filter((invoice) => invoice._id !== invoiceId),
    }))
  },
  setInvoiceStatus: (invoiceId, status) => {
    set((state) => ({
      invoices: state.invoices.map((invoice) =>
        invoice._id === invoiceId ? { ...invoice, status } : invoice
      ),
    }))
  },
  editInvoice: (invoice) => console.log('Editing invoice', invoice),
  duplicateInvoice: (invoice) => console.log('Duplicating invoice', invoice),
}))

// Component
const Invoices: React.FC = () => {
  const [filter, setFilter] = useState<string | null>(null)
  const {
    invoices,
    deleteInvoice,
    setInvoiceStatus,
    editInvoice,
    duplicateInvoice,
  } = useInvoiceStore()

  useEffect(() => {
    const confirmedDeleteHandler = (
      event: Electron.IpcRendererEvent,
      index: number,
      invoiceId: string
    ) => {
      if (index === 0) {
        deleteInvoice(invoiceId)
      }
    }

    ipc.on('confirmed-delete-invoice', confirmedDeleteHandler)

    return () => {
      ipc.removeListener('confirmed-delete-invoice', confirmedDeleteHandler)
    }
  }, [deleteInvoice])

  const handleDeleteInvoice = (invoiceId: string) => {
    createDialogWindow(
      {
        type: 'warning',
        title: 'Delete Invoice',
        message: 'Are you sure you want to delete this invoice?',
        buttons: ['Yes', 'No'],
      },
      'confirmed-delete-invoice',
      invoiceId
    )
  }

  const handleSetFilter = (event: React.MouseEvent<HTMLButtonElement>) => {
    const newFilter = event.currentTarget.dataset.filter!
    setFilter((currentFilter) =>
      currentFilter === newFilter ? null : newFilter
    )
  }

  const filteredInvoices = filter
    ? invoices.filter((invoice) => invoice.status === filter)
    : invoices

  const statuses = ['paid', 'pending', 'refunded', 'cancelled']

  return (
    <PageWrapper>
      <PageHeader>
        <PageHeaderTitle>Invoices</PageHeaderTitle>
        <PageHeaderActions>
          <i className="ion-funnel" />
          <ButtonsGroup>
            {statuses.map((status) => (
              <Button
                key={status}
                active={filter === status}
                data-filter={status}
                onClick={handleSetFilter}
              >
                {status.charAt(0).toUpperCase() + status.slice(1)}
              </Button>
            ))}
          </ButtonsGroup>
        </PageHeaderActions>
      </PageHeader>
      <PageContent bare>
        {invoices.length === 0 ? (
          <Message info text="No invoices available." />
        ) : (
          <div className="row">
            {filteredInvoices.map((invoice, index) => (
              <Invoice
                key={invoice._id}
                dateFormat="YYYY-MM-DD"
                deleteInvoice={handleDeleteInvoice}
                duplicateInvoice={duplicateInvoice}
                editInvoice={editInvoice}
                setInvoiceStatus={setInvoiceStatus}
                index={index}
                invoice={invoice}
              />
            ))}
          </div>
        )}
      </PageContent>
    </PageWrapper>
  )
}

export default _withFadeInAnimation(Invoices)
