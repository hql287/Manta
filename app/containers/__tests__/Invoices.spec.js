import React from 'react';
import { render, shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

// Component
import {
  PageWrapper,
  PageHeader,
  PageHeaderTitle,
  PageHeaderActions,
  PageContent,
} from '../../components/shared/Layout';
import { Invoices } from '../Invoices';
import Button, { ButtonsGroup } from '../../components/shared/Button';

// Mocks
const dispatch = jest.fn();
const t = jest.fn();
const invoices = [
  {
    _id: 'first-invoice',
    status: 'pending',
  },
  {
    _id: 'second-invoice',
    status: 'refunded',
  },
  {
    _id: 'third-invoice',
    status: 'paid',
  },
  {
    _id: 'fourth-invoice',
    status: 'cancelled',
  },
];
jest.mock('../../components/invoices/Invoice', () => () => (
  <div className="invoice" />
));

jest.mock('../../components/shared/Message', () => () => (
  <div className="message" />
));

describe('render component correctly', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<Invoices t={t} invoices={invoices} dispatch={dispatch} />);
  });

  it('receives correct props', () => {
    const mountWrapper = mount(
      <Invoices t={t} invoices={invoices} dispatch={dispatch} />
    );
    expect(mountWrapper.prop('t')).toEqual(t);
    expect(mountWrapper.prop('dispatch')).toEqual(dispatch);
    expect(mountWrapper.prop('invoices')).toEqual(invoices);
  });

  it('render correct number of invoices', () => {
    expect(wrapper.find('.invoice')).toHaveLength(invoices.length);
  });

  it('render empty message when there is no invoice in the DB', () => {
    const newWrapper = mount(
      <Invoices t={t} invoices={[]} dispatch={dispatch} />
    );
    expect(newWrapper.find('.message')).toHaveLength(1);
  });

  it('render a correct page layout', () => {
    expect(wrapper.find(PageWrapper)).toHaveLength(1);
    expect(wrapper.find(PageHeader)).toHaveLength(1);
    expect(wrapper.find(PageHeaderTitle)).toHaveLength(1);
    expect(wrapper.find(PageHeaderActions)).toHaveLength(1);
    expect(wrapper.find(PageContent)).toHaveLength(1);
    expect(wrapper.find(PageContent)).toHaveLength(1);
  });

  it('renders filter with four options', () => {
    const HeaderActions = wrapper.find(PageHeaderActions).first();
    expect(HeaderActions.find(Button)).toHaveLength(4);
  });

  it('matches snapshot', () => {
    const tree = renderer
      .create(<Invoices t={t} dispatch={dispatch} invoices={invoices} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('handle actions correcrtly', () => {
  let wrapper, spySetFilter, actions, filterBtn;
  beforeEach(() => {
    spySetFilter = jest.spyOn(Invoices.prototype, 'setFilter');
    wrapper = mount(
      <Invoices t={t} invoices={invoices} dispatch={dispatch} />
    );
    actions = wrapper.find(PageHeaderActions).first();
    filterBtn = actions.find(Button).last();
    filterBtn.simulate('click', {
      target: {
        dataset: {
          filter: 'paid',
        },
      },
    });
  });

  it('set correct filter', () => {
    expect(spySetFilter).toHaveBeenCalled();
    const filter = wrapper.state().filter;
    expect(filter).toEqual('paid');
    expect(filter).not.toEqual('pending');
  });

  it('display only invoices with matched status', () => {
    const filter = wrapper.state().filter;
    const filteredInvoices = invoices.filter(
      invoice => invoice.status === filter
    );
    expect(wrapper.find('.invoice')).toHaveLength(filteredInvoices.length);
  });
});
