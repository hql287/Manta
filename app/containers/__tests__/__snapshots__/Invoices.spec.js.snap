// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render component correctly matches snapshot 1`] = `
<div
  className="Layout__PageWrapperStyle-iMhhJx jVLYWQ"
>
  <div
    className="Layout__PageHeaderStyle-iGolPb jBPoCI"
  >
    <p
      className="Layout__PageHeaderTitleStyle-dWMwjR hsLeeO"
    />
    <div
      className="Layout__PageHeaderActionsStyle-eysiui fAXKb"
    >
      <i
        className="ion-funnel"
      />
      <div
        className="Button__ButtonsGroupStyle-kpiclP cdhXmK"
      >
        <button
          className="Button__ButtonStyle-coFSWz coAhhp"
          data-filter="paid"
          onClick={[Function]}
        >
          paid
        </button>
        <button
          className="Button__ButtonStyle-coFSWz coAhhp"
          data-filter="pending"
          onClick={[Function]}
        >
          pending
        </button>
        <button
          className="Button__ButtonStyle-coFSWz coAhhp"
          data-filter="refunded"
          onClick={[Function]}
        >
          refunded
        </button>
        <button
          className="Button__ButtonStyle-coFSWz coAhhp"
          data-filter="cancelled"
          onClick={[Function]}
        >
          cancelled
        </button>
      </div>
    </div>
  </div>
  <div
    className="Layout__PageContentStyle-dERDcF ciKDVW"
  >
    <div
      className="row"
    >
      <div
        className="col-lg-6"
      >
        <div
          className="Invoice__Wrapper-cjNJPS coWKLW"
        >
          <div
            className="Invoice__StatusBar-jBWtSa joKiG"
          />
          <div
            className="Invoice__Header-cGSpHV gmDFcR"
          >
            <div
              className="Invoice__Status-fxIELu bbALRd"
            >
              <span>
                <i
                  className="ion-loop"
                />
              </span>
            </div>
            <div
              className="Invoice__ButtonsGroup-PwNbx emusdh"
            >
              <button
                className="Button__ButtonLinkStyle-caVwCA imXwRc"
                onClick={[Function]}
              >
                <i
                  className="ion-ios-copy"
                />
              </button>
              <button
                className="Button__ButtonLinkStyle-caVwCA imXwRc"
                onClick={[Function]}
              >
                <i
                  className="ion-trash-a"
                />
              </button>
            </div>
          </div>
          <div
            className="Invoice__Body-eBItTh igYqzw"
          >
            <div
              className="Invoice__Row-hFNBjq cQdnct"
            >
              <div
                className="Invoice__Field-cZpgXH fDrpWO"
              >
                <label />
                <h2>
                  Penelope Hettinger
                </h2>
              </div>
            </div>
            <div
              className="Invoice__Row-hFNBjq cQdnct"
            >
              <div
                className="Invoice__Field-cZpgXH fDrpWO"
              >
                <label />
                <p>
                  first-in
                </p>
              </div>
              <div
                className="Invoice__Field-cZpgXH fDrpWO"
              >
                <label />
                <p>
                  USD
                   
                  3,843.00
                   
                </p>
              </div>
            </div>
            <div
              className="Invoice__Row-hFNBjq cQdnct"
            >
              <div
                className="Invoice__Field-cZpgXH fDrpWO"
              >
                <label />
                <p>
                  2018-03-21T18:42:56+07:00
                </p>
              </div>
              <div
                className="Invoice__Field-cZpgXH fDrpWO"
              >
                <label />
                <p />
              </div>
            </div>
          </div>
          <div
            className="Invoice__Footer-dcHDEd cJiGTY"
          >
            <div>
              <div
                className="SplitButton__Wrapper-jsRMIo gVGdJM"
              >
                <div
                  className="SplitButton__ButtonsGroup-hdzsBq dhTQNh"
                >
                  <a
                    href="#"
                    onClick={[Function]}
                  />
                  <a
                    href="#"
                    onClick={[Function]}
                  >
                    <i
                      className="ion-arrow-down-b"
                    />
                  </a>
                </div>
              </div>
            </div>
            <button
              className="Button__ButtonStyle-coFSWz coAhhp"
              onClick={[Function]}
            />
            <button
              className="Button__ButtonStyle-coFSWz coAhhp"
              onClick={[Function]}
            />
          </div>
        </div>
      </div>
      <div
        className="col-lg-6"
      >
        <div
          className="Invoice__Wrapper-cjNJPS coWKLW"
        >
          <div
            className="Invoice__StatusBar-jBWtSa ePnrEY"
          />
          <div
            className="Invoice__Header-cGSpHV gmDFcR"
          >
            <div
              className="Invoice__Status-fxIELu bhzHkr"
            >
              <span>
                <i
                  className="ion-arrow-return-left"
                />
              </span>
            </div>
            <div
              className="Invoice__ButtonsGroup-PwNbx emusdh"
            >
              <button
                className="Button__ButtonLinkStyle-caVwCA imXwRc"
                onClick={[Function]}
              >
                <i
                  className="ion-ios-copy"
                />
              </button>
              <button
                className="Button__ButtonLinkStyle-caVwCA imXwRc"
                onClick={[Function]}
              >
                <i
                  className="ion-trash-a"
                />
              </button>
            </div>
          </div>
          <div
            className="Invoice__Body-eBItTh igYqzw"
          >
            <div
              className="Invoice__Row-hFNBjq cQdnct"
            >
              <div
                className="Invoice__Field-cZpgXH fDrpWO"
              >
                <label />
                <h2>
                  Penelope Hettinger
                </h2>
              </div>
            </div>
            <div
              className="Invoice__Row-hFNBjq cQdnct"
            >
              <div
                className="Invoice__Field-cZpgXH fDrpWO"
              >
                <label />
                <p>
                  second-i
                </p>
              </div>
              <div
                className="Invoice__Field-cZpgXH fDrpWO"
              >
                <label />
                <p>
                  USD
                   
                  3,843.00
                   
                </p>
              </div>
            </div>
            <div
              className="Invoice__Row-hFNBjq cQdnct"
            >
              <div
                className="Invoice__Field-cZpgXH fDrpWO"
              >
                <label />
                <p>
                  2018-03-21T18:42:56+07:00
                </p>
              </div>
              <div
                className="Invoice__Field-cZpgXH fDrpWO"
              >
                <label />
                <p />
              </div>
            </div>
          </div>
          <div
            className="Invoice__Footer-dcHDEd cJiGTY"
          >
            <div>
              <div
                className="SplitButton__Wrapper-jsRMIo gVGdJM"
              >
                <div
                  className="SplitButton__ButtonsGroup-hdzsBq dhTQNh"
                >
                  <a
                    href="#"
                    onClick={[Function]}
                  />
                  <a
                    href="#"
                    onClick={[Function]}
                  >
                    <i
                      className="ion-arrow-down-b"
                    />
                  </a>
                </div>
              </div>
            </div>
            <button
              className="Button__ButtonStyle-coFSWz coAhhp"
              onClick={[Function]}
            />
            <button
              className="Button__ButtonStyle-coFSWz coAhhp"
              onClick={[Function]}
            />
          </div>
        </div>
      </div>
      <div
        className="col-lg-6"
      >
        <div
          className="Invoice__Wrapper-cjNJPS coWKLW"
        >
          <div
            className="Invoice__StatusBar-jBWtSa SnHNo"
          />
          <div
            className="Invoice__Header-cGSpHV gmDFcR"
          >
            <div
              className="Invoice__Status-fxIELu lkqwu"
            >
              <span>
                <i
                  className="ion-checkmark"
                />
              </span>
            </div>
            <div
              className="Invoice__ButtonsGroup-PwNbx emusdh"
            >
              <button
                className="Button__ButtonLinkStyle-caVwCA imXwRc"
                onClick={[Function]}
              >
                <i
                  className="ion-ios-copy"
                />
              </button>
              <button
                className="Button__ButtonLinkStyle-caVwCA imXwRc"
                onClick={[Function]}
              >
                <i
                  className="ion-trash-a"
                />
              </button>
            </div>
          </div>
          <div
            className="Invoice__Body-eBItTh igYqzw"
          >
            <div
              className="Invoice__Row-hFNBjq cQdnct"
            >
              <div
                className="Invoice__Field-cZpgXH fDrpWO"
              >
                <label />
                <h2>
                  Penelope Hettinger
                </h2>
              </div>
            </div>
            <div
              className="Invoice__Row-hFNBjq cQdnct"
            >
              <div
                className="Invoice__Field-cZpgXH fDrpWO"
              >
                <label />
                <p>
                  third-in
                </p>
              </div>
              <div
                className="Invoice__Field-cZpgXH fDrpWO"
              >
                <label />
                <p>
                  USD
                   
                  3,843.00
                   
                </p>
              </div>
            </div>
            <div
              className="Invoice__Row-hFNBjq cQdnct"
            >
              <div
                className="Invoice__Field-cZpgXH fDrpWO"
              >
                <label />
                <p>
                  2018-03-21T18:42:56+07:00
                </p>
              </div>
              <div
                className="Invoice__Field-cZpgXH fDrpWO"
              >
                <label />
                <p />
              </div>
            </div>
          </div>
          <div
            className="Invoice__Footer-dcHDEd cJiGTY"
          >
            <div>
              <div
                className="SplitButton__Wrapper-jsRMIo gVGdJM"
              >
                <div
                  className="SplitButton__ButtonsGroup-hdzsBq dhTQNh"
                >
                  <a
                    href="#"
                    onClick={[Function]}
                  />
                  <a
                    href="#"
                    onClick={[Function]}
                  >
                    <i
                      className="ion-arrow-down-b"
                    />
                  </a>
                </div>
              </div>
            </div>
            <button
              className="Button__ButtonStyle-coFSWz coAhhp"
              onClick={[Function]}
            />
            <button
              className="Button__ButtonStyle-coFSWz coAhhp"
              onClick={[Function]}
            />
          </div>
        </div>
      </div>
      <div
        className="col-lg-6"
      >
        <div
          className="Invoice__Wrapper-cjNJPS coWKLW"
        >
          <div
            className="Invoice__StatusBar-jBWtSa ttyPp"
          />
          <div
            className="Invoice__Header-cGSpHV gmDFcR"
          >
            <div
              className="Invoice__Status-fxIELu eMUCie"
            >
              <span>
                <i
                  className="ion-backspace"
                />
              </span>
            </div>
            <div
              className="Invoice__ButtonsGroup-PwNbx emusdh"
            >
              <button
                className="Button__ButtonLinkStyle-caVwCA imXwRc"
                onClick={[Function]}
              >
                <i
                  className="ion-ios-copy"
                />
              </button>
              <button
                className="Button__ButtonLinkStyle-caVwCA imXwRc"
                onClick={[Function]}
              >
                <i
                  className="ion-trash-a"
                />
              </button>
            </div>
          </div>
          <div
            className="Invoice__Body-eBItTh igYqzw"
          >
            <div
              className="Invoice__Row-hFNBjq cQdnct"
            >
              <div
                className="Invoice__Field-cZpgXH fDrpWO"
              >
                <label />
                <h2>
                  Penelope Hettinger
                </h2>
              </div>
            </div>
            <div
              className="Invoice__Row-hFNBjq cQdnct"
            >
              <div
                className="Invoice__Field-cZpgXH fDrpWO"
              >
                <label />
                <p>
                  fourth-i
                </p>
              </div>
              <div
                className="Invoice__Field-cZpgXH fDrpWO"
              >
                <label />
                <p>
                  USD
                   
                  3,843.00
                   
                </p>
              </div>
            </div>
            <div
              className="Invoice__Row-hFNBjq cQdnct"
            >
              <div
                className="Invoice__Field-cZpgXH fDrpWO"
              >
                <label />
                <p>
                  2018-03-21T18:42:56+07:00
                </p>
              </div>
              <div
                className="Invoice__Field-cZpgXH fDrpWO"
              >
                <label />
                <p />
              </div>
            </div>
          </div>
          <div
            className="Invoice__Footer-dcHDEd cJiGTY"
          >
            <div>
              <div
                className="SplitButton__Wrapper-jsRMIo gVGdJM"
              >
                <div
                  className="SplitButton__ButtonsGroup-hdzsBq dhTQNh"
                >
                  <a
                    href="#"
                    onClick={[Function]}
                  />
                  <a
                    href="#"
                    onClick={[Function]}
                  >
                    <i
                      className="ion-arrow-down-b"
                    />
                  </a>
                </div>
              </div>
            </div>
            <button
              className="Button__ButtonStyle-coFSWz coAhhp"
              onClick={[Function]}
            />
            <button
              className="Button__ButtonStyle-coFSWz coAhhp"
              onClick={[Function]}
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
